### PR TITLE
Add Ruby 3.1.0

### PR DESCRIPTION
**Pull request by shmokmt**
Related to https://github.com/ruby/ruby-builder/pull/8 🎅 
the above PR must merge first to pass the tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,7 +86,7 @@ jobs:
       matrix:
         os: [ubuntu-latest, macos-latest]
         # Due to https://github.com/actions/runner/issues/849, we have to use quotes for '3.0'
-        ruby: [2.5, 2.6, 2.7, '3.0', head, jruby, jruby-head, truffleruby, truffleruby-head]
+        ruby: [2.5, 2.6, 2.7, '3.0', 3.1, head, jruby, jruby-head, truffleruby, truffleruby-head]
```
Comment (eregon) [inline README.md:89]: Could you remove 2.5 from this example since it's EOL now?

Comment (eregon) [inline README.md:89]:
```suggestion
        ruby: [2.6, 2.7, '3.0', 3.1, head, jruby, jruby-head, truffleruby, truffleruby-head]
```

**Review by eregon — APPROVED** [review]
Thank you!

**Comment by lopopolo** [comment]
@eregon it looks like Ruby 3.1.0 isn't available for Windows. Should I file a ticket or will that be automatically picked up once the installer is built?

**Comment by MSP-Greg** [comment]
> automatically picked up once the installer is built?

It's not automatic, but it should be added here shortly after it's built.

**Pull request merged**